back/app/controllers/web_api/v1/ideas_controller.rb

```diff
@@ -209,21 +208,20 @@ def update
     params[:idea][:phase_ids] ||= [] if params[:idea].key?(:phase_ids)
     params_service.mark_custom_field_values_to_clear!(input.custom_field_values, params[:idea][:custom_field_values])
 
-    user_can_moderate_project = UserRoleService.new.can_moderate_project?(project, current_user)
+    user_can_moderate_project = UserRoleService.new.can_moderate_project?(input.project, current_user)
     update_params = idea_params(input.custom_form, user_can_moderate_project).to_h
     update_params[:custom_field_values] = params_service.updated_custom_field_values(input.custom_field_values, update_params[:custom_field_values])
     CustomFieldService.new.compact_custom_field_values! update_params[:custom_field_values]
-    input.assign_attributes update_params
-    authorize input
-    if anonymous_not_allowed?(phase)
-      render json: { errors: { base: [{ error: :anonymous_participation_not_allowed }] } }, status: :unprocessable_entity
-      return
-    end
-    if idea_status_not_allowed?(input)
-      render json: { errors: { idea_status_id: [{ error: 'Cannot manually assign inputs to an automatic status', value: input.idea_status_id }] } }, status: :unprocessable_entity
-      return
+
+    ActiveRecord::Base.transaction do # Assigning relationships cause database changes
+      input.assign_attributes update_params
+      authorize input
+      if not_allowed_update_errors(input)
+        render json: not_allowed_update_errors(input), status: :unprocessable_entity
+        return # rubocop:disable Rails/TransactionExitStatement
```

Comment:
Rubocop does not recommend using `return` in a transaction, because in previous Rails versions, this would not rollback the transaction, whereas now we do get a rollback. The alternative would have been raising an exception and then catch it outside of the transaction, but Koen and I agreed it's less effort and more readable to just disable the cop here.